exp/addons/api/v1beta1/clusterresourcesetbinding_types.go

```diff
@@ -58,14 +58,22 @@ type ResourceSetBinding struct {
 
 // IsApplied returns true if the resource is applied to the cluster by checking the cluster's binding.
 func (r *ResourceSetBinding) IsApplied(resourceRef ResourceRef) bool {
+	resourceBinding := r.GetResourceBinding(resourceRef)
+	if resourceBinding == nil {
+		return false
+	}
+
+	return resourceBinding.Applied
+}
+
+// GetResourceBinding returns a ResourceBinding for a resource ref if present.
+func (r *ResourceSetBinding) GetResourceBinding(resourceRef ResourceRef) *ResourceBinding {
```

Comment:
nit
```suggestion
func (r *ResourceSetBinding) GetResource(resourceRef ResourceRef) *ResourceBinding {
```
(binding is implicit in the type the method belongs to)